### PR TITLE
Various C file decomp

### PR DESCRIPTION
**Pull request by Rainchus**

### CODE REVIEW

File: src/world/area_nok/nok_04/9EDBF0.c [hunk 1]
```diff
@@ -1,30 +1,244 @@
...
+    /* 0x20 */ f32 unk_20;
+    /* 0x24 */ f32 unk_24;
+    /* 0x28 */ f32 unk_28;
+    /* 0x2C */ f32 unk_2C;
+    /* 0x30 */ f32 unk_30;
+    /* 0x34 */ f32 unk_34;
+    /* 0x38 */ f32 unk_38;
+    /* 0x3C */ f32 unk_3C;
+    /* 0x40 */ f32 unk_40;
+} Unk80241440_2; //sizeof 0x44
```
Comment (nanaian) [inline src/world/area_nok/nok_04/9EDBF0.c:21]:
```suggestion
} Unk80241440_2; // size = 0x44
```

File: src/world/area_nok/nok_04/9EDBF0.c [hunk 2]
```diff
@@ -1,30 +1,244 @@
...
+    /* 0x24 */ f32 unk_24;
+    /* 0x28 */ f32 unk_28;
+    /* 0x2C */ f32 unk_2C;
+    /* 0x30 */ f32 unk_30;
+    /* 0x34 */ f32 unk_34;
+    /* 0x38 */ f32 unk_38;
+    /* 0x3C */ f32 unk_3C;
+    /* 0x40 */ Unk80241440_2* unk_40;
+    /* 0x44 */ Unk80241440_2* unk_44;
+} Unk80241440; //sizeof 0x48
```
Comment (nanaian) [inline src/world/area_nok/nok_04/9EDBF0.c:42]:
```suggestion
} Unk80241440; // size = 0x48
```

File: src/world/partner/lakilester.c [hunk 3]
```diff
@@ -8,14 +9,81 @@ extern s32 D_802BFF00;
...
+extern s32 D_802BFF24;
+extern f32 D_802BFF28;
+
+extern s16 D_8010C97A;
 
 s32 func_802BD7DC(void);
+s32 partner_use_ability(void);
+f32 get_player_normal_pitch(void);
+void partner_kill_ability_script(void);
+f64 fabs(f64 temp);
```
Comment (nanaian) [inline src/world/partner/lakilester.c:26]: I'd be surprised if this wasn't declared already? (by libultra)

File: src/world/partner/lakilester.c [hunk 4]
```diff
@@ -100,33 +170,389 @@ ApiStatus func_802BD2D4_320E24(Evt* script, s32 isInitialCall) {
...
 }
 
-INCLUDE_ASM(s32, "world/partner/lakilester", func_802BD678_3211C8);
+void func_802BD678_3211C8(Npc* npc) {
+    unkPartnerStruct* temp_v1;
 
-INCLUDE_ASM(s32, "world/partner/lakilester", func_802BD6BC_32120C);
+    if (D_8010C954 != 0) {
+        temp_v1 = D_802BFE7C_3239CC;
+        D_8010C954 = 0;
```
Comment (nanaian) [inline src/world/partner/lakilester.c:178]:
```suggestion
        D_8010C954 = NULL;
```

File: src/world/partner/lakilester.c [hunk 5]
```diff
@@ -100,33 +170,389 @@ ApiStatus func_802BD2D4_320E24(Evt* script, s32 isInitialCall) {
     return ApiStatus_BLOCK;
 }
 
-INCLUDE_ASM(s32, "world/partner/lakilester", func_802BD678_3211C8);
+void func_802BD678_3211C8(Npc* npc) {
+    unkPartnerStruct* temp_v1;
 
-INCLUDE_ASM(s32, "world/partner/lakilester", func_802BD6BC_32120C);
+    if (D_8010C954 != 0) {
```
Comment (nanaian) [inline src/world/partner/lakilester.c:176]: This is a pointer
```suggestion
    if (D_8010C954 != NULL) {
```

File: src/world/partner/lakilester.c [hunk 6]
```diff
@@ -100,33 +170,389 @@ ApiStatus func_802BD2D4_320E24(Evt* script, s32 isInitialCall) {
...
+    f32 temp_f24 = gPartnerActionStatus.stickX;
+    f32 temp_f26 = gPartnerActionStatus.stickY;
+    f32 temp_f22 = -temp_f26;
+    f32 atan = atan2(0.0f, 0.0f, temp_f24, temp_f22);
+    f32 temp = clamp_angle(atan + gCameras->currentYaw);
+    f32 phi_f20 = 0.0f;
+    
+    if (dist2D(0.0f, 0.0f, temp_f24, temp_f22) >= 1.0) {
+        phi_f20 = 3.0f;
+        if ((temp_f24 * temp_f24) + (temp_f26 * temp_f26) > 3025.0f) {
```
Comment (nanaian) [inline src/world/partner/lakilester.c:195]:
```suggestion
        if (SQ(temp_f24) + SQ(temp_f26) > 3025.0f) {
```

File: src/world/partner/lakilester.c [hunk 7]
```diff
@@ -100,33 +170,413 @@ ApiStatus func_802BD2D4_320E24(Evt* script, s32 isInitialCall) {
...
+    f32 hitRx, hitRz;
+    f32 outLength;
+    f32 outX, outY, outZ;
+    Npc *npc = get_npc_unsafe(NPC_PARTNER);
+    s32 colliderTypeID;
+    s32 raycastResult;
+    s32 ret;
+
+    if (playerStatus->animFlags & 0x20000000) {
+        playerStatus->animFlags &= 0xDFFFFFFF;
```
[the remark this answers is not in the document]
Comment (nanaian) [reply src/world/partner/lakilester.c:217]: this is unresolved

File: src/world/partner/lakilester.c [hunk 8]
```diff
@@ -100,33 +170,389 @@ ApiStatus func_802BD2D4_320E24(Evt* script, s32 isInitialCall) {
...
+    npc->unk_84 = -1;
+    npc->jumpScale += D_802BFEF8;
+
+    if (npc->jumpScale > 12.0f) {
+        npc->jumpScale = 12.0f;
+    }
+}
+#else
+INCLUDE_ASM(s32, "world/partner/lakilester", func_802BDDD8_321928);
+#endif
```
Comment (nanaian) [inline src/world/partner/lakilester.c:532]: damn. good effort

File: src/world/partner/lakilester.c [hunk 9]
```diff
@@ -245,10 +671,131 @@ ApiStatus func_802BF4F0_323040(Evt* script, s32 isInitialCall) {
...
-INCLUDE_ASM(s32, "world/partner/lakilester", func_802BFB44_323694);
+void func_802BFAA8_3235F8(Npc* npc) {
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
 
-INCLUDE_ASM(s32, "world/partner/lakilester", func_802BFBA0_3236F0);
+    if (partnerActionStatus->actionState.b[1] != 0) {
+        if (D_802BFF0C != 0) {
+            *npc = partnerActionStatus->npc;
+            gGameStatusPtr->unk_7D = 1;
+            set_action_state(0x21);
```
Comment (nanaian) [inline src/world/partner/lakilester.c:697]: Can you use the actionstate enum

File: src/world/partner/lakilester.c [hunk 10]
```diff
@@ -245,10 +671,131 @@ ApiStatus func_802BF4F0_323040(Evt* script, s32 isInitialCall) {
...
+                temp_f0 = playerStatus->position.x;
+                npc->pos.x = temp_f0;
+                npc->moveToPos.x = temp_f0;
+                temp_f4 = playerStatus->position.z;
+                npc->pos.z = temp_f4;
+                npc->moveToPos.z = temp_f4;
+                playerStatus->position.y = npc->pos.y + 10.0f;
+                partner_kill_ability_script();
+            } else {
+                set_action_state(0x21);
```
Comment (nanaian) [inline src/world/partner/lakilester.c:742]: Use enum

File: src/world/partner/lakilester.c [hunk 11]
```diff
@@ -245,10 +671,131 @@ ApiStatus func_802BF4F0_323040(Evt* script, s32 isInitialCall) {
...
+            playerStatus->position.y = npc->pos.y + 10.0f;
+            playerStatus->position.z = npc->pos.z;
+            playerStatus->targetYaw = npc->yaw;
+            func_802BFB44_323694(2.0f);
+            script->functionTemp[1] -= 1;
+            
+            if (script->functionTemp[1] == 0) {
+                if (script->varTable[12] != 0) {
+                    partnerActionStatus->actionState.b[1] = tempVar;
+                    set_action_state(0x21);
```
Comment (nanaian) [inline src/world/partner/lakilester.c:789]: Use enum

**Review by nanaian — CHANGES_REQUESTED** [review]
Hopefully the last review!

File: src/759b0_len_61b0.c [hunk 12]
```diff
@@ -57,9 +57,57 @@ void npc_get_slip_vector(f32* outX, f32* outZ, f32 aX, f32 aZ, f32 bX, f32 bZ) {
...
+    hitDepth = temp_f22;
+    aZ = length * inverseOutCosTheta;
+    raycastHitID = npc_raycast_general(ignoreFlags, *x - (radius * outSinTheta * 0.5f), *y,
+                                        *z - (radius * inverseOutCosTheta * 0.5f), outSinTheta,
+                                        0.0f, inverseOutCosTheta, &hitX, &hitY, &hitZ, &hitDepth, &bX, &hitNy, &bZ);
+    phi_s5 = 0;
+
+    if ((raycastHitID >= 0) && (hitDepth <= temp_f22)) {
+        temp_f0 = atan2(0.0f, 0.0f, sqrtf( (bX * bX) + (bZ * bZ) ), -hitNy);
+        if ((temp_f0 > 60.0f) && (temp_f0 < 90.0f)) {
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/759b0_len_61b0.c:90]: still unnecessary parens for the tempf0 greater than / less than conditions

**Review by nanaian — APPROVED** [review]

**Pull request merged**